XenAdmin/Diagnostics/Checks/AssertCanEvacuateCheck.cs

```diff
@@ -253,6 +229,25 @@ private Problem GetProblem(IXenConnection connection, XenRef<VM> vmRef, string[]
             }
         }
 
+        private static CannotMigrateVM.CannotMigrateVMReason GetMoreSpecificReasonForCannotMigrateVm(VM vm, CannotMigrateVM.CannotMigrateVMReason reason)
+        {
+            var gm = vm.Connection.Resolve(vm.guest_metrics);

```

Comment:
This could go into the first if block to avoid resolving unecessarily, even if it's not an expensive operation, but it's ok, you don't have to change it.